drivers/sensor/lsm303dlhc_magn/lsm303dlhc_magn.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2018 Philémon Jaermann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <i2c.h>
+#include <init.h>
+#include <sensor.h>
+#include <logging/sys_log.h>
+
+#include "lsm303dlhc_magn.h"
+
+static int lsm303dlhc_sample_fetch(struct device *dev,
+				   enum sensor_channel chan)
+{
+	struct lsm303dlhc_magn_data *drv_data = dev->driver_data;
+	u8_t tries = 11;
+	u8_t magn_buf[6];
+	u8_t status;
+
+	/* Check data ready flag */
+	while (tries-- > 0) {
+		if (i2c_reg_read_byte(drv_data->i2c,
+				      LSM303DLHC_MAGN_I2C_ADDR,
```

Comment:
Similar to the accel, get the i2c addr from a const device config struct.

Reply:
I'll do it asap.

Reply:
Done